hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -899,11 +898,13 @@ public String getMultipartKeyFSO(String volume, String bucket, String key, Strin
     final long volumeId = getVolumeId(volume);
     final long bucketId = getBucketId(volume,
             bucket);
-    long parentId =
-            OMFileRequest.getParentID(volumeId, bucketId, key, this);
-
-    String fileName = OzoneFSUtils.getFileName(key);
+    final String nonFSOMultipartKey =
+        getMultipartKey(volume, bucket, key, uploadId);
+    final OmMultipartKeyInfo multipartKeyInfo =
+        getMultipartInfoTable().get(nonFSOMultipartKey);
```

Comment:
Please handle null `OmMultipartKeyInfo` if the entry does not exist in `multipartInfoTable` to prevent NPE that might crash OM since `RuntimeException` is not caught in `validateAndUpdateCache`.

Reply:
Sure